### DISCUSSION TITLE
API command in basic

### DISCUSSION THREAD

**Discussion by erlendnagel** [paulscottrobson/neo6502-firmware#279] [Implemented Ideas]
Especially for those API functions that do not have a basic equivalent and perhaps never will, it would improve the readability of the basic programs a lot if commands like the below existed
APIWAIT  ( "while peek($FF00) <> 0:wend")
API(2,13) ( "poke($FF01),13: poke ($FF00),2 ")

For optimal readability a programmer could even add "console = 2 : getcursor = 13" and then just use API(console,getcursor) and peek the results

**Comment by paulscottrobson** [comment]
Not a bad idea. Maybe wait as a parameter apicall 2,13, true. This doesn't really hide much of the reality though ; there is setting up of values in the message buffer, error checking and retrieval.  It's easy enough to do as a procedure

**Comment by erlendnagel** [comment]
True defining a procedure to handle it is easy. Let's keep the basic lean :-)
